### PR TITLE
Fix LK build path for catalyst shared library

### PR DESCRIPTION
**Pull request by josephleekl**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [ ] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [ ] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
This PR fixes Lightning Kokkos with Catalyst failing due to wrong build path supplied to `setup.py` and lightning kokkos in editable mode.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-77123]

### CODE REVIEW

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.py [hunk 1]
```diff
@@ -534,7 +534,7 @@ def get_c_interface():
         #   lib.<system>-<architecture>-<python-id>"
         # To avoid mismatching the folder name, we search for the shared object instead.
         # TODO: locate where the naming convention of the folder is decided and replicate it here.
-        editable_mode_path = package_root.parent.parent / "build"
+        editable_mode_path = package_root.parent.parent / "build_lightning_kokkos"
```
Comment (maliasadi) [inline pennylane_lightning/lightning_kokkos/lightning_kokkos.py:537]: Shall we add a unit test for this editable_mode path?

File: setup.py [hunk 2]
```diff
@@ -161,7 +161,7 @@ def build_extension(self, ext: CMakeExtension):
             if platform.system() in ["Linux", "Darwin"]:
                 shared_lib_ext = {"Linux": ".so", "Darwin": ".dylib"}[platform.system()]
                 source = os.path.join(f"{extdir}", f"lib{backend}_catalyst{shared_lib_ext}")
-                destination = os.path.join(os.getcwd(), "build")
+                destination = os.path.join(os.getcwd(), self.build_temp)
```
Comment (maliasadi) [inline setup.py:164]: :1st_place_medal:

**Review by maliasadi — APPROVED** [review]
Thanks @josephleekl! Please add a changelog entry for this PR before merge :)

File: pennylane_lightning/lightning_kokkos/lightning_kokkos.py (hunk 1 above)
Comment (multiphaseCFD) [reply pennylane_lightning/lightning_kokkos/lightning_kokkos.py:537]: Second to it.

Comment (josephleekl) [reply pennylane_lightning/lightning_kokkos/lightning_kokkos.py:537]: I'll add one!

File: .github/workflows/tests_lkcpu_python.yml [hunk 3]
```diff
@@ -282,9 +282,112 @@ jobs:
...
+          PL_BACKEND=${{ matrix.pl_backend }} python scripts/configure_pyproject_toml.py
+          CMAKE_ARGS="-DCMAKE_PREFIX_PATH=${{ github.workspace }}/Kokkos" python -m pip install -e . --config-settings editable_mode=compat -vv
+      
+      - name: Install ML libraries for interfaces
+        run: |
+          python -m pip install --upgrade torch==$TORCH_VERSION -f https://download.pytorch.org/whl/cpu/torch_stable.html
+          python -m pip install --upgrade "jax[cpu]"  # This also installs jaxlib
+          python -m pip install --upgrade tensorflow~=$TF_VERSION keras~=$TF_VERSION
+
+      - name: Run PennyLane-Lightning unit tests
```
Comment (multiphaseCFD) [inline .github/workflows/tests_lkcpu_python.yml:359]: Thanks @josephleekl . Would the logic [here](https://github.com/PennyLaneAI/pennylane-lightning/blob/b6e308e77278e64959782204f11716196ee605e7/.github/workflows/tests_gpu_python.yml#L210-L221) be simpler?

Comment (josephleekl) [reply .github/workflows/tests_lkcpu_python.yml]: That's much more straightforward, thanks!

**Review by multiphaseCFD — APPROVED** [review]
Happy to approve! Thanks @josephleekl !

**Pull request merged**